static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts

```diff
@@ -346,70 +331,89 @@ const filterEmptyValues = (pageTargets: Record<string, unknown>) => {
 	return filtered;
 };
 
-const rebuildPageTargeting = () => {
-	latestCmpHasInitialised = cmp.hasInitialised();
-	const adConsentState = getAdConsentFromState(latestCMPState);
-	const ccpaState = latestCMPState?.ccpa
-		? latestCMPState.ccpa.doNotSell
-		: null;
-	const tcfv2EventStatus = latestCMPState?.tcfv2
-		? latestCMPState.tcfv2.eventStatus
-		: 'na';
+const getConsentRelatedPageTargeting = (
+	canTargetAds: boolean,
+	consentState?: ConsentState,
+): PageTargeting => {
+	const amtgrp = consentState?.tcfv2
+		? getAdManagerGroup(canTargetAds)
+		: getAdManagerGroup();
+
+	if (!consentState) {
+		return {
+			amtgrp,
+			cmp_interaction: 'na',
+			consent_tcfv2: 'na',
+			pa: 'f',
+			rdp: 'na',
+		};
+	}
+	return {
+		amtgrp,
+		cmp_interaction: consentState.tcfv2
+			? consentState.tcfv2.eventStatus
+			: 'na',
+		consent_tcfv2: consentState.tcfv2
+			? consentedToAllPurposes(consentState.tcfv2.consents)
+				? 't'
+				: 'f'
+			: 'na',
+		pa: canTargetAds ? 't' : 'f',
+		rdp: consentState.ccpa
+			? consentState.ccpa.doNotSell
+				? 't'
+				: 'f'
+			: 'na',
+	};
+};
 
+const getPageTargeting = (consentState?: ConsentState): PageTargeting => {
```

Comment:
As I understand it, there's just one place in the source that we don't pass consent state in, and that's when we first use it in `prepare-googletag.ts`? 

Do you think having consent state as an optional parameter make it possible to forget to pass in the consent in other places in the code? I wonder if it's worth making it more difficult to not pass in consent state. Or maybe this just isn't that much of a problem 🤷

Reply:
> I agree we should make it hard to do the wrong thing. I was hoping to look into the appnexus issue as a follow up to this, so all going well we won't leave that hole open for long.

Awesome, one for the follow-up then.

> If we can't get around it, perhaps we just have two functions that make it explicit e.g. getPageTargeting(consentState: ConsentState) and getPageTargetingWithoutConsentState() ?

Good idea!